cmake/modules/dts.cmake

```diff
@@ -349,17 +349,24 @@ endif()
 #
 # Run GEN_DTS_CMAKE_SCRIPT.
 #
+# A temporary file is used copied to the original file if it differs, this prevents a cycle when
+# sysbuild is used of configuring and building multiple times due to the dts.cmake file of images
+# having a newer modification time than the sysbuild build.ninja file, despite the output having
+# not changed
```

Comment:
In general this is not only tied to sysbuild.

Touching any file during CMake configure time can trigger additional build steps later.
Build steps which may be unnecessary if there are no changes to the configure time generated file.

This may not only be related to dts.cmake, but can be true for other files created at configure time.
Therefore this is not fixing a sysbuild issue, sysbuild just happens to be impacted more by this fact than other parts of the system.

Long term a more general handling of such cases would be nice so that we don't regenerate a file in the first place instead of having to generate, compare and move.